test/internal/laws/Setoid.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var forall = require('./forall');
+
+
+module.exports = forall({
+
+  //  a `equals` a = true
+  reflexivity: function(a) {
+    return a.equals(a);
+  },
+
+  //  a `equals` b = b `equals` a
+  symmetry: function(a, b) {
+    return a.equals(b) === b.equals(a);
+  },
+
+  //  a `equals` b & b `equals` c => a `equals` c
+  transitivity: function(a, b, c) {
+    return !a.equals(b) || !b.equals(c) || a.equals(c);
```

Comment:
I think it takes fewer moments to understand version using ternary, then with just boolean logic. So I would wrote for `a.equals(b) && b.equals(c) ? a.equals(c) : true`. It's also direct translation of the `if/else` expression you wrote above, which is more descriptive and easier to understand. it reads as: `if a equals b and b equals c then a must equal c` which is not the case for other variations